hazelcast/src/main/java/com/hazelcast/config/PartitioningAttributeConfig.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.internal.config.ConfigDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+
+/**
+ * Contains the configuration for attributes used to create
+ * {@link com.hazelcast.partition.strategy.AttributePartitioningStrategy}
+ */
+public class PartitioningAttributeConfig implements IdentifiedDataSerializable {
+    private String attributeName;
+
+    public PartitioningAttributeConfig() { }
+
+    public PartitioningAttributeConfig(final PartitioningAttributeConfig config) {
```

Comment:
+1

Reply:
Checking the list, besides dynamic config, which we'll do later, we lack these: 
- ConfigXmlGenerator, ConfigXmlGeneratorTest
- XmlOnlyConfigBuilderTest#testConfig2Xml2FullConfig and XmlOnlyConfigBuilderTest#testConfig2Xml2FullConfig_withAdvancedNetworkConfig and XmlOnlyConfigBuilderTest#testConfig2Xml2DefaultConfig)
- full configs:

    hazelcast-fullconfig-without-network.[xml&yaml] (included into the two below)
    hazelcast-fullconfig.[xml&yaml]
    hazelcast-fullconfig-advanced-network-config.[xml&yaml]
-

Reply:
I believe ConfigXmlGenerator, ConfigXmlGeneratorTest don't handle map configs, its delegated to the DynamicConfigXmlGenerator, which have the test [here](https://github.com/hazelcast/hazelcast/pull/23535/files#diff-009b995eebc4057bc80bfc93b7da25a3a58fc0d7642a286a12781bc53b8d1287R288-R305).

I pushed the fullconfig changes, entire test suite of XmlOnlyConfigBuilderTest passes, including those listed tests (checked in debugger specifically to see whether configs include partitioning attributes - they do, so that's tested).